### ISSUE TITLE
Steam does not connect on kernels newer than 2019-06-17

### ISSUE THREAD

**Issue by killerrabbit** [ValveSoftware/steam-for-linux#6326]
#### Your system information

* Steam client version (build number or date): Latest, can't determine exact version
* Distribution (e.g. Ubuntu): Debian 9 Stable
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

When starting steam, after logging in and authenticating past steam guard, it hangs on "Connecting to steam account: (account name)." Sometimes it will change to "Updating Steam information" with an empty progress bar.

After a while, it will give a dialog box saying that says "Could not connect to Steam network."

Output of console: https://gist.github.com/killerrabbit/94419ec85564e248c044d21e0fd961b3

#### Steps for reproducing this issue:

1. Install Steam
2. Attempt to log in

**Comment by jonlap** [comment]
I have this same problem in Arch Linux with the exception that I am opted into Steam client beta. This issue started few days ago, out of nowhere, on installation where Steam client has been working OK before. 

I have removed my ~/.steam and ~/.local/share/Steam and tried again with no luck. I also tested with flatpak current stable, same results. Meanwhile, Steam (flatpak stable) is running fine on my laptop with Void Linux. Both of these machines are in same network.

[Console output](https://gist.github.com/jonlap/9f2d3ee20eebc946bd03811c5b263451)

**Comment by shirona** [comment]
Having a very similar issue here on NixOS after the latest system update. Main difference is, in my case, it manages to log in around 10% of the times. Keep trying to log in, maybe even kill the process while it is booting so it attempts to check the integrity of the installation (it probably is irrelevant, but the times it has worked for me, this was the case).

**Comment by blastermaster77** [comment]
Same problem with ubuntu 19.04 after update, Could not connect to Steam network. I tried on my laptop and samething sad...

**Comment by revbo** [comment]
I am also experiencing this problem on both Arch Linux and Kubuntu 19.04. Both OS run on the same exact hardware. Meanwhile, a Windows 10 PC on the same network works fine with Steam.

On both Kubuntu and Arch Linux, I was able to login and receive a Steam Guard password through email to authenticate on a new machine. At this point the error occurs. 

[Here is my steam output](https://gist.github.com/revbo/377d9ca5e023ab2c1e731f8f411df87b) from the Arch Linux installation. This is a brand-new, fresh install of Arch Linux.

**Comment by iswritergeek** [comment]
I also have this problem on my Arch install. Been happening for two days now, ever since I rolled over to the new kernel. There also seems to be a problem with the Breeze plasma theme and the way the new kernel does networking, as can be seen [here in my console output](https://gist.github.com/revbo/377d9ca5e023ab2c1e731f8f411df87b).

I'm running the arch linux-lts, btw. I've had no problems connecting up till today using the steam package.

**Comment by killerrabbit** [comment]
I downgraded my kernel from 4.9.0-9 to 4.9.0-8 and steam is now working correctly.

Something in the kernel update broke it.

**Comment by mark-blum** [comment]
I had this same problem on my Arch Linux as well, but I was able to actually start steam by directly launching a game I had installed, with the command `steam steam://rungameid/440` (which launches Team Fortress 2). I would test this further but after this I seem to be able to launch steam normally without any issues.

**Comment by Xanderful** [comment]
Same issue here, steam started fine until I purged and re-installed for another issue on Ubuntu 19.04 vanilla edition. Same computer, steam on Widnows starts fine and on another edition of Ubuntu running Mate 19.04 same. I am running 5.0.0-17-generic on the version that is not working.

**Comment by hartmunt** [comment]
I have the same problem running Ubuntu 19.04. The issue apparently started after the upgrade to kernel version 5.0.0-17-generic a few days ago for me as well.

I maybe have another insight though:

I'm running a strict multi-user setup which basically means that every major application has its own user or user group (leaving some details out). This also applies to steam in my case.
One of the first thing I tried was starting steam from my main desktop user, since the actual steam installation is user-specific in the respective $HOME folder.
Steam downloaded its bootstrap or whatever and asked me to login and provide the Steam Guard Email authentication thingy.
And it worked. And it continues to work consistently with that user.

This of course means that i have no games or custom configuration in that steam instance but Steam at least starts. It seems like a complete purge and re-installation could possibly fix the problem, but I'm not sure about that. I really don't want to reinstall Steam and/or the Steam games for my steam-user and I'm not sure that would even work, so I'm hoping there will be another solution.

**Comment by Seegras** [comment]
Same here on Debian Unstable with mainline (not Debian) kernel 5.1.12 (and 5.1.11 also). NVidia drivers 430.26. Login works, but after that I get "Could not connect to Steam network.". I also get that when starting it with the parameter "-tcp"

My assumption is that some kernel security-upgrade, which was important enough to get backported might be the trigger for steam not working anymore.

**Comment by blastermaster77** [comment]
The weird thign is that both my laptop and desktop have the latest updates and one connects now and the desktop does not. Yesterday both could not connect to steam network. I dont know what is going on. Hope the fix it soon, its driving me nutz!
> Same here on Debian Unstable with mainline (not Debian) kernel 5.1.12 (and 5.1.11 also). NVidia drivers 430.26. Login works, but after that I get "Could not connect to Steam network.". I also get that when starting it with the parameter "-tcp". And I also deleted steam (except steamapps) and reinstalled. No avail.
> 
> My assumption is that some security-upgrade, probably in the kernel, which was important enough to get backported might be the trigger for steam not working anymore.
> 
> I think it's the networking in the kernel:
> 
> > linux (4.9.168-1+deb9u3) stretch-security; urgency=high
> > [ Salvatore Bonaccorso ]
> > 
> > * tcp: limit payload size of sacked skbs (CVE-2019-11477)
> > * tcp: tcp_fragment() should apply sane memory limits (CVE-2019-11478)
> > * tcp: add tcp_min_snd_mss sysctl (CVE-2019-11479)
> > * tcp: enforce tcp_min_snd_mss in tcp_mtu_probing()
> > * tcp: fix fack_count accounting on tcp_shift_skb_data()
> > 
> > [ Ben Hutchings ]
> > 
> > * tcp: Avoid ABI change for DoS fixes
> > * mm/mincore.c: make mincore() more conservative (CVE-2019-5489)
> > * brcmfmac: add length checks in scheduled scan result handler
> > * brcmfmac: assure SSID length from firmware is limited (CVE-2019-9500)
> > * brcmfmac: add subtype check for event handling in data path (CVE-2019-9503)
> > * tty: mark Siemens R3964 line discipline as BROKEN (CVE-2019-11486)
> > * coredump: fix race condition between mmget_not_zero()/get_task_mm() and
> >   core dumping (CVE-2019-11599)
> > * net: rds: force to destroy connection if t_sock is NULL in
> >   rds_tcp_kill_sock(). (CVE-2019-11815) (Closes: #928989)
> > * ext4: zero out the unused memory region in the extent tree block
> >   (CVE-2019-11833)
> > * Bluetooth: hidp: fix buffer overflow (CVE-2019-11884)
> > * mwifiex: Fix possible buffer overflows at parsing bss descriptor
> >   (CVE-2019-3846)
> > * mwifiex: Abort at too short BSS descriptor element
> > * mwifiex: Don't abort on small, spec-compliant vendor IEs
> > * mwifiex: Fix heap overflow in mwifiex_uap_parse_tail_ies()
> >   (CVE-2019-10126)
> > 
> > Sun, 16 Jun 2019 15:38:39 +0100

**Comment by blastermaster77** [comment]
I copied the .steam folder from my laptop that does login and it fixed the problem this is weird.

**Comment by jkaart** [comment]
I have same problem with fresh installed 64-bit Ubuntu 18.04.

**Comment by jonlap** [comment]
Downgraded kernel from 5.1.11-arch1-1-ARCH to 5.1.8-arch1-1-ARCH (which is the latest previous version I had in pacman cache) and Steam now works again. Changed nothing else so yeah, something kernel related.

**Comment by Seegras** [comment]
Valve, please fix steam. Going back to a kernel which has at least 13 known security vulnerabilites (all the changes denoted with CVE's are security fixes) is not an option.

**Comment by yevheniypopov** [comment]
Running steam from terminal with -tcp param fixed this problem for me:
```
$steam -tcp
```

**Comment by dhollinger** [comment]
I'm on Pop!_OS 19.04 (desktop optimized Ubuntu) with a 5.x kernel and haven't had any problems with steam. Been on this kernel for a few weeks now.

**Comment by Seegras** [comment]
It's not just a 5.x kernel, but probably 5.1.11 or newer (or an earlier one with some security fixes backported). Apparently 5.1.8 works, 5.1.9 and 5.1.10 might also.

**Comment by XakepSDK** [comment]
Linux arch 5.1.11-arch1-1-ARCH
Steam: Jun 17 2019, at 23:31:27
Steam works fine.
Also 1 or 2 days earlier, steam asked me to log-in manually.

**Comment by ugly95** [comment]
I'm glad people are starting to notice this.  It's been happening to me since Monday when I got the 5.0.0-17 kernel update and I hadn't seen anyone comment on it and thought I was the only one.

As others have said, rolling back to an earlier kernel version works (in my case to 5.0.0-16) resolves this issue.

My console output ends at 

```
Running Steam on linuxmint 19.1 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
[2019-06-18 19:48:15] Startup - updater built Jun 17 2019 23:31:08
[2019-06-18 19:48:15] Verifying installation...
[2019-06-18 19:48:15] Verification complete
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
[2019-06-18 19:48:41] Shutdown
```
The google results for issues with not being able to connect suggest deleting ~/.steam, which resulting in Steam re-downloading and installing and successfully connecting.  But after closing Steam and trying to login again, it starts failing again.

Another suggestion was to run `steam --reset`.  Not sure if it's related, but I get `Error: Couldn't find bootstrap, it's not safe to reset Steam. Please contact technical support.`

Running Steam through Wine also connects successfully.

**Comment by XakepSDK** [comment]
Linux arch 5.1.12-arch1-1-ARCH
Works fine! Note: i didn't log out, but just exit / start steam.

**Comment by blastermaster77** [comment]
I can confirm that rolling back to an older kernel fixed the problem, in my case I install 5.0.0-13-generic
on ubuntu 19.04 if I update to the newest kernel 5.0.0-17 the problem start again, waiting for a fix thanks. I also noticed that when it is installing the update on the new kernel it stops and it does not install it, hope that helps.

**Comment by niarbeht** [comment]
It's happening to me whether I use `steam -tcp` or not.  `steam -tcp` does not appear to fix the issue on Arch with kernel `5.1.12-arch1-1-ARCH`.  I have tried `steam --reset` and deleting the contents of `~/.local/share/steam`

**Comment by alosarjos** [comment]
Are we sure it's something kernel related? Haven't had any issues in the last weeks and I check for updates daily.

Arch Linux x86_64 / Kernel 5.1.12-zen1-1-zen
Steam beta client

¿You guys have any kind of network config on your machines?

**Comment by SergeyLatyshev** [comment]
After installing 5.1.11 (and 5.1.12 today) with ukuu, I have this problem too. Steam is in autorun, and never connect at the first try, only after I relaunch it. Sometimes it takes several tries to launch it successfully.

**Comment by darkpumpkins** [comment]
Have you tried into a terminal  ?
export STEAM_RUNTIME=1 
steam
it solved my issue on my ubuntu 19.04.

**Comment by emmecii** [comment]
Can this issue happen with the HWE kernels of Ubuntu 18.04?

**Comment by dennis1248** [comment]
I have a very similar issue which recently appeared and is most likely related, Steam struggles to connect to the Steam friends network after it has started. But it does manage to connect after 10-20 seconds and a retry.
 
I use Void Linux btw, I believe the issue first appeared on kernel version 5.1.11. Currently running 5.1.12 with the same issue.

**Comment by iFloody** [comment]
Can confirm, kernel 5.0.0-13-generic works for me, but 5.0.0-17 don't. And the strange thing is: When I'm on kernel 5.0.0-17 I can't login on my main steam account (fails on updating steam information) but when I try to login on a empty account it works just fine. (Ubuntu 19.04)

**Comment by mirh** [comment]
If you have some spare time, https://wiki.archlinux.org/index.php/Bisecting_bugs_with_Git

**Comment by MaxXor** [comment]
No issues for me on `Linux 5.1.12-arch1-1-ARCH #1 SMP PREEMPT Wed Jun 19 09:16:00 UTC 2019 x86_64 GNU/Linux`.

**Comment by debianxfce** [comment]
No Steam problems with the latest AMD wip kernel :
  Host: ryzenpc Kernel: 5.2.0-rc2+ x86_64 bits: 64 Desktop: Xfce 4.13.2 
  Distro: Debian GNU/Linux 10 (buster) 

Never Steam or other problems with this kernel.

**Comment by blastermaster77** [comment]
Do you clean install steam? If you don't you won't see the problem.

**Comment by AngryPenguinPL** [comment]
Works for me on OpenMandriva Cooker with kernel 5.1.12. Tested on two PC and notebook.

**Comment by flinterson** [comment]
Did a clean build of Void Linux on a different laptop. Void uses kernel 4.19.53, which has the new tcp stack changes. Had the same issue out of the box

**Comment by Markaos** [comment]
Ubuntu 19.04, kernel 5.0.0-17-generic: the login works randomly for me - Steam completely refused to start twice, then started but with Friends offline, then connected to Friends after I clicked the Reconnect (or whatever the text is) button. Browser and Steam Chat app on the same network worked just fine, steamstat.us reported no problems, so it should be this problem

**Comment by jonlap** [comment]
> Void uses kernel 4.19.53, which has the new tcp stack changes. Had the same issue out of the box

I have maybe few weeks old Void install, kernel 4.19.51_1, and clean install of Steam. Steam is working (for me at least) with this kernel version. Anyone have time to bisect as mirh already suggested?

**Comment by JayGuerette** [comment]
Fedora 30, kernel 5.0.9-301 and 5.0.16-300. Steam works, but streaming stopped working. Tried multiple computers. It executes the game on the remote computer, but displays a black screen and after a bit says "couldn't connect to remote computer"

**Comment by parasense** [comment]
Fedora 30, fully updated as of Fri Jun 21 2019  using distro kernel "5.1.11-300.fc30.x86_64" works perfectly fine. I'm able to start the Steam client, launch Team Fortress 2, and play the game like normal. No issues what so ever. Tested using an Intel NUC Hades Canyon (NUC8i7HVK) if that helps.

**Comment by garpu** [comment]
I'm using 4.19.53, and no issues connecting to Steam.  (Slackware current.)  Specific kernel config that's causing the issue, as well?

**Comment by jonlap** [comment]
> Specific kernel config that's causing the issue, as well?

I guess it's possible, although there are people with same kernel version and same distro (and thus same kernel config) as I have reporting Steam works for them. Far fetched but maybe it's hardware/driver specific in addition to kernel version?

**Comment by Chiitoo** [comment]
Same config, probably, /if/ they didn't build their own kernel.  :]

This may be even farther fetched, but... it isn't IPv6 related again is it?  (Probably not.)

I, too, am using the 5.1.11 Linux kernel currently, on Gentoo, and I have IPv6 disabled everywhere (including the kernel) since my ISP does not provide it at all still.

Steam works for me, though I feel like the start-up time is longer than before (past 40 seconds).  I also did have a single failure while connecting some days ago, but I forget if it was before or after switching to 5.1.11... might be completely coincidental.

**Comment by jonlap** [comment]
> This may be even farther fetched, but... it isn't IPv6 related again is it? (Probably not.)

I tried disabling IPv6 at some point, didn't help.

**Comment by urbenlegend** [comment]
I think the latest Steam Client update fixed this already? I have 3 Fedora machines currently. Two of them I use very often for gaming and I had already updated them to the latest Steam Client prior to upgrading them to kernel 5.1.11. Steam logged in just fine on these two machines.
Later, I tried to login into Steam on my 3rd machine, a laptop. This one I don't use for gaming very often, and I had upgraded to kernel 5.1.11 before updating Steam. Steam failed to login consistently on this machine. I then ran `steam --tcp` which allowed me to login and update. After the update, the problem no longer occurs.

**Comment by dearlygparted** [comment]
I haven't had any issues connected to Steam, and I've been upgrading my kernel regularly. Currently running this kernel:

```
$ uname -r
5.1.12-arch1-1-ARCH
```
What am I missing?

**Comment by sgtnasty** [comment]
I had the same problem yesterday on neon (18.04) and decided to install ubuntu 19.04 and I get the same issue.
First I installed steam from the repos, then I uninstalled it and downloaded the deb package from steampowered.com and still get the issue.

**Comment by Farcrada** [comment]
I run WireGuard at work to tunnel home (`wg-quick up xyz`), if I turn it off I can instantly and consistently log into Steam. Latest updates of everything (fresh Steam install too). All relevant places to get Steam have the same result; Software Manager, packages (terminal `apt install steam`) and the `.deb` on the website.

When reconnecting WireGuard once logged into Steam, downloads aren't working. Steam friends-chat still receives and (seems to) send messages, thus I still appear online. Starting a new install doesn't even start `Create local game files...`; just hangs.

Linux Mint 19.1 (Cinnamon)
Kernel: `4.15.0-52-generic`

**Comment by evcz** [comment]
Seems something similar could be affecting csgo servers (on centos 7) where external http calls using steam libraries seems to be randomly affected too when the payload is big resulting in fragmented packets

**Comment by blastermaster77** [comment]
Just asking, did you fresh install steam or are you using a previous
installation? The problem manifests when you delete all steam installation
with the newest kernel with the tcp stack security update.

**Comment by BrosephJ** [comment]
Running Arch with 5.1.12-zen1-1-zen and Steam through LSI ("Use native runtime" disabled because Steam webhelper bug). No connection issues here.

**Comment by mwpow3ll** [comment]
I don't see this being a kernel issue; I'm running F29 5.1.11 and works fine. Previously was running 5.1.8, and 5.0.9 with no issues as well.

Sounds more like a runtime issue... with those affected.

**Comment by AngryPenguinPL** [comment]
> I don't see this being a kernel issue; I'm running F29 5.1.11 and works fine. Previously was running 5.1.8, and 5.0.9 with no issues as well.
> 
> Sounds more like a runtime issue... with those affected.

I have similar feelings. Today I tried to reproduce it on my computer but without any luck. All kernels 5.1.9, 5.1.10, 5.1.11 and 5.1.12 on OpenMandriva Cooker and steam working fine. Even on fresh OS instalation I not see this issue...

**Comment by evcz** [comment]
> > I don't see this being a kernel issue; I'm running F29 5.1.11 and works fine. Previously was running 5.1.8, and 5.0.9 with no issues as well.
> > Sounds more like a runtime issue... with those affected.
> 
> I have similar feelings. Today I tried to reproduce it on my computer but without any luck. All kernels 5.1.9, 5.1.10, 5.1.11 and 5.1.12 on OpenMandriva Cooker and steam working fine. Even on fresh OS instalation I not see this issue...

try ubuntu

**Comment by mwpow3ll** [comment]
> try ubuntu

I guess you don't want help? ok. Why don't you try Fedora... see, I can do it too.

**Comment by joshuahunt** [comment]
If you are seeing this issue can you check a new counter that was added as part of the SACK fixes to see if it's non-zero? You can check the counter by doing:
# nstat -az | grep -i wqueue
TcpExtTCPWqueueTooBig           0                  0.0

**Comment by flinterson** [comment]
@joshuahunt Here's the output from my laptop that has this problem:

`TcpExtTCPWqueueTooBig           61                 0.0`

**Comment by mwpow3ll** [comment]
F29 running 5.1.11:

TcpExtTCPWqueueTooBig           0                  0.0

**Comment by joshuahunt** [comment]
> F29 running 5.1.11:
> 
> TcpExtTCPWqueueTooBig 0 0.0

Are you seeing the connectivity issue?

**Comment by mwpow3ll** [comment]
> > F29 running 5.1.11:
> > TcpExtTCPWqueueTooBig 0 0.0
> 
> Are you seeing the connectivity issue?

No

**Comment by joshuahunt** [comment]
> > > F29 running 5.1.11:
> > > TcpExtTCPWqueueTooBig 0 0.0
> > 
> > 
> > Are you seeing the connectivity issue?
> 
> No; Just trying to provide some context.

Got it. Appreciate it. Thanks. :)

**Comment by norbusan** [comment]
I am having issues, Debian unstable, 5.1.12 (same with .11)
```
[~] nstat -az | grep -i wqueue
TcpExtTCPWqueueTooBig           28                 0.0
```

**Comment by torvalds** [comment]
> 
> When starting steam, after logging in and authenticating past steam guard, it hangs on "Connecting to steam account: (account name)." Sometimes it will change to "Updating Steam information" with an empty progress bar.
> 
> After a while, it will give a dialog box saying that says "Could not connect to Steam network."

If anybody is set to build their own kernel, can you try the patch from here:

    https://patchwork.ozlabs.org/patch/1120222/

to see if it fixes your issue?

Thanks.

**Comment by Dexter9313** [comment]
Not sure why nor how, but it broke on my Kubuntu 18.04 and got fix by several reinstalls then steam --reset ... (It didn't work the first time, I had to reinstall steam to be able to issue the steam --reset command.)

**Comment by ugly95** [comment]
I am on Linux Mint 19.1 using kernel 5.0.0-17 and I am having this issue:

Before trying to start Steam:
```
nstat -az | grep -i wqueue
TcpExtTCPWqueueTooBig           0                  0.0
```
After Steam fails to connect:
```
nstat -az | grep -i wqueue
TcpExtTCPWqueueTooBig           6                  0.0
```

**Comment by niarbeht** [comment]
> > When starting steam, after logging in and authenticating past steam guard, it hangs on "Connecting to steam account: (account name)." Sometimes it will change to "Updating Steam information" with an empty progress bar.
> > After a while, it will give a dialog box saying that says "Could not connect to Steam network."
> 
> If anybody is set to build their own kernel, can you try the patch from here:
> 
> ```
> https://patchwork.ozlabs.org/patch/1120222/
> ```
> 
> to see if it fixes your issue?
> 
> Thanks.

I just applied this patch to `5.1.12.arch1-1`, and I can connect to Steam again, both using Steam runtime and using native runtime.  Thank you for the prompt assistance.

**Comment by Plagman** [comment]
Can also confirm the patch fixes it for me, after building a kernel that reproduces the problem.

**Comment by flinterson** [comment]
I was also able to build a patched `5.1.12.arch1-1` and that got it working. Thank you for the patch, Linus.

**Comment by shoober420** [comment]
On Arch with 5.1.12, I can login to Steam, but Steam games seem to be unable to download updates. CSGO keeps telling me that there’s an update available at the top when launching the game, but doesn’t proceed to download the update.

I assume using the -tcp parameter would fix this, as it would force all traffic onto TCP instead of UDP, but I’ll just wait till this gets patched officially.

I feel the only reason I got farther than others is the fact my sysctl.conf file is tweaked to all hell.

sysctl.conf - https://pastebin.com/raw/s0F3CbRT

**Comment by Rehaell** [comment]
For me it is the same, both on my Ubuntu 18.04 instalations (Desktop + Laptop).

Ubuntu 18.04
Kernel: 4.18.0-22-generic

The only wait it connects is using the flag tcp, as in steam -tcp.

**Comment by torvalds** [comment]
> > 
> > If anybody is set to build their own kernel, can you try the patch from here:
> > ```
> > https://patchwork.ozlabs.org/patch/1120222/
> > ```
> > 
> > 
> > to see if it fixes your issue?
> > Thanks.
> 
> I just applied this patch to `5.1.12.arch1-1`, and I can connect to Steam again, both using Steam runtime and using native runtime. 

Ok, it's in my tree now, and on the way to -stable, and thence to distro kernels. So it should all be fixed in distro kernels soon.

**Comment by niarbeht** [comment]
> > > If anybody is set to build their own kernel, can you try the patch from here:
> > > ```
> > > https://patchwork.ozlabs.org/patch/1120222/
> > > ```
> > > 
> > > 
> > > to see if it fixes your issue?
> > > Thanks.
> > 
> > 
> > I just applied this patch to `5.1.12.arch1-1`, and I can connect to Steam again, both using Steam runtime and using native runtime.
> 
> Ok, it's in my tree now, and on the way to -stable, and thence to distro kernels. So it should all be fixed in distro kernels soon.
> 
> Commit b6653b3629e5 ("tcp: refine memory limit test in tcp_fragment()") upstream in case anybody cares.

Playing Mechwarrior, don't care :P

**Comment by Seegras** [comment]
> https://patchwork.ozlabs.org/patch/1120222/
Patch works. Thank you.

**Comment by lilianmoraru** [comment]
https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.1.14

**Comment by nalquas** [comment]
> For me it is the same, both on my Ubuntu 18.04 instalations (Desktop + Laptop).
> 
> Ubuntu 18.04
> Kernel: 4.18.0-22-generic
> 
> The only wait it connects is using the flag tcp, as in steam -tcp.

I'm also not able to connect to steam; I have tried reinstalling Steam completely (apt purge, delete ~/.steam and ~/.local/share/steam), but cannot get Steam to work. The tcp-flag isn't working for me, either.

KDE Neon (Ubuntu 18.04)
Kernel: 4.18.0-22-generic

**Comment by darkpumpkins** [comment]
> For me it is the same, both on my Ubuntu 18.04 instalations (Desktop + Laptop).
> 
> Ubuntu 18.04
> Kernel: 4.18.0-22-generic
> 
> The only wait it connects is using the flag tcp, as in steam -tcp.

Same here unlike what I said earlier, steam -tcp did help me to launch steam.
kernel version : 5.0.0-18-generic

**Comment by nalquas** [comment]
> > For me it is the same, both on my Ubuntu 18.04 instalations (Desktop + Laptop).
> > Ubuntu 18.04
> > Kernel: 4.18.0-22-generic
> > The only wait it connects is using the flag tcp, as in steam -tcp.
> 
> Same here unlike what I said earlier, steam -tcp did help me to launch steam.
> kernel version : 5.0.0-18-generic
I am only affected on my desktop, but not my laptop. Both have the following setup:
KDE Neon (Ubuntu 18.04)
Kernel-Version: 4.18.0-22-generic

The -tcp flag does not seem to help me connect.

**Comment by sgtnasty** [comment]
> The -tcp flag does not seem to help me connect.

I had to use:

`steam --tcp`

**Comment by DoogeJ** [comment]
Same issue here on a fresh Ubuntu 19.04 installation.

Kernel is `5.0.0-17-generic`, starting with `-tcp` or `--tcp` doesn't work for me, I can confirm that `TcpExtTCPWqueueTooBig` is increasing.

Console output from Steam starting:
```Running Steam on ubuntu 19.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Gtk-Message: 23:19:10.382: Failed to load module "gail"
Gtk-Message: 23:19:10.382: Failed to load module "atk-bridge"

(steam:16939): Gtk-WARNING **: 23:19:10.386: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/main.rc:775: error: unexpected identifier 'direction', expected character '}'

(steam:16939): Gtk-WARNING **: 23:19:10.388: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
Installing breakpad exception handler for appid(steam)/version(1560817063)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
[0622/231910.734645:INFO:crash_reporting.cc(242)] Crash reporting enabled for process: browser
[0622/231910.750181:WARNING:crash_reporting.cc(281)] Failed to set crash key: UserID with value: 0
[0622/231910.750240:WARNING:crash_reporting.cc(281)] Failed to set crash key: BuildID with value: 1559604844
[0622/231910.750245:WARNING:crash_reporting.cc(281)] Failed to set crash key: SteamUniverse with value: Public
[0622/231910.750249:WARNING:crash_reporting.cc(281)] Failed to set crash key: Vendor with value: Valve
/usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so: undefined symbol: g_log_structured_standard
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
CAppInfoCacheReadFromDiskThread took 0 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
CApplicationManagerPopulateThread took 0 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)

** (steam:16939): WARNING **: 23:19:10.890: Unknown device type 14

** (steam:16939): WARNING **: 23:19:10.890: Could not create object for /org/freedesktop/NetworkManager/Devices/1: unknown object type
Installing breakpad exception handler for appid(steam)/version(1560817063)
Opted-in Controller Mask for AppId 0: 0

** (steam:16939): WARNING **: 23:19:11.115: Ignoring invalid property 'autoconnect-priority'

** (steam:16939): WARNING **: 23:19:11.115: Unknown setting 'proxy'

** (steam:16939): WARNING **: 23:19:11.115: Ignoring invalid property 'address-data'

** (steam:16939): WARNING **: 23:19:11.115: Ignoring invalid property 'route-data'

** (steam:16939): WARNING **: 23:19:11.115: Ignoring invalid property 'route-data'

** (steam:16939): WARNING **: 23:19:11.115: Ignoring invalid property 'address-data'
Installing breakpad exception handler for appid(steam)/version(1560817063)

(steam:16939): Gtk-WARNING **: 23:19:11.628: gtk_disable_setlocale() must be called before gtk_init()
Installing breakpad exception handler for appid(steam)/version(1560817063)
Installing breakpad exception handler for appid(steam)/version(1560817063)
Exiting workitem thread
[2019-06-22 23:19:10] Startup - updater built Jun 17 2019 23:31:08
[2019-06-22 23:19:10] Verifying installation...
[2019-06-22 23:19:10] Verification complete
[2019-06-22 23:19:33] Shutdown
```

**Comment by codedcosmos** [comment]
-tcp worked for me

Ubuntu 19.04
5.0.0-17-generic

**Comment by TechtonicSoftware** [comment]
On arch and can confirm updating to Kernel 5.1.14 from the testing repo fixes the issue (you will need to use an up-to-date mirror at time of posting).

**Comment by Rehaell** [comment]
For those not having success with the -tcp flag.

My laptop was running ubuntu 18.04 lts and kernel 4.18.0-22-generic and it ran fine with the flag.
My desktop was running ubuntu 18.04 lts with kernel 4.15 and it did not ran with the flag, so I just updated the kernel to 4.18 with ukuu and now runs fine with the flag.

Not sure if this is the smartest way to do it, but it worked for me.

**Comment by Farcrada** [comment]
> I run WireGuard at work to tunnel home (`wg-quick up xyz`), if I turn it off I can instantly and consistently log into Steam. Latest updates of everything (fresh Steam install too). All relevant places to get Steam have the same result; Software Manager, packages (terminal `apt install steam`) and the `.deb` on the website.
> 
> When reconnecting WireGuard once logged into Steam, downloads aren't working. Steam friends-chat still receives and (seems to) send messages, thus I still appear online. Starting a new install doesn't even start `Create local game files...`; just hangs.
> 
> Linux Mint 19.1 (Cinnamon)
> Kernel: `4.15.0-52-generic`

No idea what I did (had no updates or installed new packages, just booted the client from no WireGuard and then connect WireGuard to already connected with WireGuard), but everything is fine now. Found it worth mentioning. It has a significant sync issue, but that might be more related to sudden closure of games (crashes, force quits, etc.).

**Comment by nalquas** [comment]
> > > For me it is the same, both on my Ubuntu 18.04 instalations (Desktop + Laptop).
> > > Ubuntu 18.04
> > > Kernel: 4.18.0-22-generic
> > > The only wait it connects is using the flag tcp, as in steam -tcp.
> > 
> > 
> > Same here unlike what I said earlier, steam -tcp did help me to launch steam.
> > kernel version : 5.0.0-18-generic
> 
> I am only affected on my desktop, but not my laptop. Both have the following setup:
> KDE Neon (Ubuntu 18.04)
> Kernel-Version: 4.18.0-22-generic

I installed kernel 4.19.55-041955-generic. This fixed the issue for me, I can now connect to steam again without any issues.

**Comment by ipkpjersi** [comment]
I'm on Ubuntu 18.04 with kernel 4.15.0-52-generic. I have removed `~/.steam` and `~/.local/share/Steam` multiple times, I have tried `steam --reset` multiple times, and I have tried `stream -tcp` and `steam --tcp` multiple times but nothing works. I do not want to compile my own kernel so I believe I will have to wait until Ubuntu 18.04 repos get a fixed kernel version which could be whoever knows how long. Interesting.

**Comment by sgtnasty** [comment]
I had to uninstall the repo deb and download the steam deb from steam website to get it to work

**Comment by jonlap** [comment]
Tested with 5.1.14-arch1-1-ARCH and Steam is now working again. Thanks everyone!

**Comment by RedSparr0w** [comment]
Thanks @DoogeJ, That fixed it for me also :+1:

**Comment by linuxgameconsortium** [comment]
@sgtnasty What was the process you followed to get the Steam client to work?

**Comment by Emanem** [comment]
> @sgtnasty What was the process you used to get the Steam client to work?
> CLI? Purge?
> Do you have "steam-devices" installed? Or for that matter "steam:i386"?

Was able to connect ~3 hours ago, now it's refusing to connect.
Running on Ubuntu 18.04 with HWE kernel:  4.18.0-24-generic

**Comment by Emanem** [comment]
> > @sgtnasty What was the process you used to get the Steam client to work?
> > CLI? Purge?
> > Do you have "steam-devices" installed? Or for that matter "steam:i386"?
> 
> Was able to connect ~3 hours ago, now it's refusing to connect.
> Running on Ubuntu 18.04 with HWE kernel: 4.18.0-24-generic

Never mind, it worked now after some hours. Looks like the auth servers are overrun with the sale...

**Comment by Saroumane** [comment]
> For those not having success with the -tcp flag.
> 
> My laptop was running ubuntu 18.04 lts and kernel 4.18.0-22-generic and it ran fine with the flag.
> My desktop was running ubuntu 18.04 lts with kernel 4.15 and it did not ran with the flag, so I just updated the kernel to 4.18 with ukuu and now runs fine with the flag.
> 
> Not sure if this is the smartest way to do it, but it worked for me.
> 
> EDIT:
> Just updated both devices to kernel 5.1.14 and works fine without the flag...

If you use nvidia proprietary driver, how do you do after updating to 5.1.1x kernel (installed with DoogeJ link) ?
I tried sudo dkms autoinstall and sudo apt reinstall nvidia-driver-418 in recovery mode without success, I can't boot anymore with 5.1.15 kernel in graphical mode.

**Comment by Saroumane** [comment]
To avoid other people in my case (Ubuntu 19.04 + nvidia proprietary driver 418) wasting time :
List of kernel tried with ubuntu-mainline-kernel.sh  script : 
v5.0.20 and v5.0.21 => Kernel Panic at boot
v5.1.0 / v5.1.1 / v5.1.12 / v5.1.13 / v5.1.14 / v5.1.15 => can't reinstall nvidia driver. (The "apt" way does nothing, and sudo dkms -k KernelVersion fails at build)

Now I could try to mess with ppa:graphics-drivers/ppa  or official nvidia binary installer, but it usually creates more harm than good.

**Comment by fwsGonzo** [comment]
> ENDING SPOILER :
> I finally chose to downgrade to kernel 5.0.13. I know I'm losing security fixes, but at least I don't break my system even more.

That's too bad. Guess I will wait, and in the meantime try -tcp at commandline.

**Comment by RafneQ** [comment]
Hi, couple days ago I tried to install and launch Steam on fresh **Ubuntu 18.04** and during login to Steam account I got error message "Could not connect to Steam network". At first I thought this was due summer sale :) but Steam was working fine on other devices. Tried on kernels **4.18.0-22** and **4.18.0-24** and in both cases Steam was not able to connect to my account.

Unfortunately doing reset or starting with "-tcp" didn't help. The only thing that worked in my case was downgrading kernel to **4.18.0-21**. After that everything is fine.

**Comment by kisak-valve** [comment]
Good day folks, thanks to those involved in triaging this issue.

The fix for this issue has landed in [5.1.14](https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.1.14), [4.19.55](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.19.55), [4.14.131](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.14.131), [4.9.184](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.9.184), and [4.4.184](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.4.184). The issue is understood and there's nothing more to investigate on this issue report. Additional confirmations here won't expedite the fix landing in distro-maintained kernel releases.

While the Steam client is affected by this issue, the fix is not considered a critical security fix, so please give the various package maintainers some time to pick this up in their normal update cycle before giving them a friendly reminder.

**Comment by Kurgol** [comment]
This is hugely annoying because on my asus n56vm laptop running ubuntu 19.04 I get the same problems described above (and it's now kernel 5.0.0-19 so above the fix level). 
![image](https://user-images.githubusercontent.com/15245142/60331469-a7ba8200-998c-11e9-8d07-e3c3050acc98.png)

But on my xps 13 running the same distribution and kernel, on the same wifi network - the xps works fine.

I have tried completely purging and re-installing steam on the asus, including removing the .steam folder after purging.

**Comment by Kurgol** [comment]
I think 5.0.0.19 has a regression because I was getting the same cannot connect on my asus n56vm laptop running ubuntu 19.04, while my xps 13 (with a 5.0.0.20 kernel) was running it fine.

Upgraded my n56vm to 5.0.0.20 and now that works and will connect without the error.

**Comment by ldra02** [comment]
> #### Your system information
> 
>     * Steam client version (build number or date): Latest, can't determine exact version
> 
>     * Distribution (e.g. Ubuntu): Debian 9 Stable
> 
>     * Opted into Steam client beta?: No
> 
>     * Have you checked for system updates?: Yes
> 
> 
> #### Please describe your issue in as much detail as possible:
> 
> Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)
> 
> When starting steam, after logging in and authenticating past steam guard, it hangs on "Connecting to steam account: (account name)." Sometimes it will change to "Updating Steam information" with an empty progress bar.
> 
> After a while, it will give a dialog box saying that says "Could not connect to Steam network."
> 
> Output of console: https://gist.github.com/killerrabbit/94419ec85564e248c044d21e0fd961b3
> #### Steps for reproducing this issue:
> 
>     1. Install Steam
> 
>     2. Attempt to log in

System Info: 

Steam Client Version: https://imagebin.ca/v/4m9MeDy7Vflo (screen shot)
Kernel Version: 4.15.0-54-generic
Distro: Ubuntu 64b 18.04

Fresh install, installed pulse, steam, UFW (allowed pass all as a troubleshooting step, hoping it would open ports)

AMD hardware (idk if this makes a difference)

Initially was running 4.x.x.x some other kernel, newer, downgraded, and worked after that.

**Comment by ugly95** [comment]
> The fix for this issue has landed in [5.1.14](https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.1.14), [4.19.55](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.19.55), [4.14.131](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.14.131), [4.9.184](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.9.184), and [4.4.184](https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.4.184). The issue is understood and there's nothing more to investigate on this issue report. Additional confirmations here won't expedite the fix landing in distro-maintained kernel releases.

I can confirm that this is also fixed in 5.0.0-20 on Linux Mint.

**Comment by pete-h** [comment]
Fixed in Ubuntu 18.04 with Kernel 4.15.0-54

**Comment by linuxgameconsortium** [comment]
> Fixed in Ubuntu 18.04 with Kernel 4.15.0-54-generic (64bit)

Confirmed. As of June 27th.

**Comment by FJoeXXL2** [comment]
4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

Problem gone for now, "-tcp" option didn't made any difference before.

**Comment by shirona** [comment]
Running right now kernel 5.1.15 on NixOS unstable. Running Steam in TCP mode has been working ever since 5.1.11, and while the issues in the default UDP mode have been alleviated, it's still not logging in consistently. Whereas in 5.1.11 I had around a <10% chance of connecting in UDP mode, now in 5.1.15 I have around a 40% to 50% chance of logging in.

**Comment by NoXPhasma** [comment]
I'm on 5.1.16 and still unable to log in consistently over UDP.

**Comment by dope250** [comment]
Was on debian 4.9.0-9-amd64 and got the problem. Downgraded to 4.9.0-8-amd64 and it worked again.
-tcp switch didn't worked.

**Comment by draeath** [comment]
OpenSuSE Leap 15.1 with kernel 4.12.14-lp151.28.7-default also has this problem, `-tcp` argument doesn't help either.

**Comment by nosoop** [comment]
Also affects SteamCMD on the current version of Debian Buster, package `linux-image-4.19.0-5-amd64` (4.19.37-5).  Last known good version on Debian is the previous version of `linux-image-4.9.0-9-amd64` (4.9.168-1+deb9u2) on Debian Stretch, which can be pulled using the [Debian snapshot archive](https://snapshot.debian.org/).

**Comment by sizeofbool** [comment]
It looks like some sort of misconfigured anti-DDoS measures on Valve side are preventing Steam Client from connecting properly.
On my Debian 'stretch' system, kernel version is 4.19 and these connectivity problems are happen sporadically when I launch Steam Client. It complains about obviously working Internet connection, and if I restart Steam Client a few times, without changing or doing anything it will work.

**Comment by cheloide** [comment]
A workaround for **debian** is downloading [this](https://snapshot.debian.org/archive/debian/20190316T031117Z/pool/main/l/linux-signed-amd64/linux-image-4.19.0-4-amd64_4.19.28-2_amd64.deb) kernel image and installing it with dpkg:
```
wget https://snapshot.debian.org/archive/debian/20190316T031117Z/pool/main/l/linux-signed-amd64/linux-image-4.19.0-4-amd64_4.19.28-2_amd64.deb -o /tmp/linux-image-4.19.0-4-amd64_4.19.28-2_amd64.deb
sudo dpkg -i /tmp/linux-image-4.19.0-4-amd64_4.19.28-2_amd64.deb
```
I recommend installing the [kernel headers](https://snapshot.debian.org/archive/debian/20190315T212803Z/pool/main/l/linux/linux-headers-4.19.0-4-amd64_4.19.28-2_amd64.deb) first if you have nvidia drivers.

Rebooting and selecting "Advanced options" in Grub menu and boot with the newly installed  kernel image
you will be able to login to steam desktop that way.

After that you will be able to open steam when booting with the current kernel.

[Source](https://www.reddit.com/r/linux_gaming/comments/cc364d/switching_from_ubuntu_to_debian_and_debian_still/etk4zuq/?context=3)

**Comment by boltronics** [comment]
I upgraded to Buster and got smacked with this. Spent hours trying to debug - I run steam in a Firejail private directory and figured the newer version of Firejail broke it somehow. Then I discovered it would work on a clean install but fail when I added my games library directory, leading me to be even more convinced it was Firejail-related... but nope it was this.

That's really bad timing to have this hit just as Buster was made stable and everyone is upgrading.

**Comment by nosoop** [comment]
Latest Debian Buster kernel package (4.19.37-5+deb10u1) includes the upstream commit for this fix.

**Comment by aaronfranke** [comment]
I'm experiencing this issue on Ubuntu 18.04 with kernel 4.18.0-22-generic.

Does anyone know what the latest working version of the kernel is?

**Comment by boltronics** [comment]
I built the latest 5.2.2 kernel, and that's working for me... usually. It still seems to fail every 2nd or 3rd time I try to load Steam, but retrying gets it working.

Note if you use firejail there is a bug in the default profile that also causes Steam to fail to connect with the same error.

https://github.com/netblue30/firejail/issues/2860

But even with that disabled, I'm still getting occasional login issues. Not sure yet if the kernel wasn't 100% fixed, or if there are other issues with the firejail profile, or possibly even Steam itself. Some factors to consider are that I have a large game library (1768 games) and the issue doesn't seem to occur when games are not installed (and I have most installed because my Internet isn't that fast and I have lots of storage). Lots are using Proton. Also, Steam is installed in a private directory so game configurations and saves are not scattered around my actual home directory.

FWIW, I use:
```
$ cat .config/firejail/steam.local 
private ~/firejail/steam

# Required for Steam to recognise game controllers.
ignore private-dev

# Prevent Proton freezes.
# https://github.com/netblue30/firejail/issues/2860
ignore seccomp
$
```

I didn't have an issue before upgrading from Stretch to Buster.

**Comment by RafneQ** [comment]
> I'm experiencing this issue on Ubuntu 18.04 with kernel 4.18.0-22-generic.
> 
> Does anyone know what the latest working version of the kernel is?

@aaronfranke for Ubuntu 18.04 it looks like it **was fixed** in **4.18.0-25** (I'm currently using this one on my machine and Steam runs just fine). It appears the affected kernels where 4.18.0-22 and 4.18.0-24 (version 23 was not published in repos)

**Comment by jamesdolan** [comment]
Hitting this issue with 4.18.0-25-generic on ubuntu 18.04, if user-land software like steam is sensitive to kernel versions that seems REALLY bad, since this issue would just keep coming back. So my guess is kernel version is a red herring, or, I at least hope to god it is.

Seems to happen on the second launch... first run of steam does the typical initial update/install, then steam loads as expected, if I close steam and try and re-launch it, I hit this issue, nuking my ~/.steam directory and launching again will re-install and run, but that's a pretty horrible workaround. So, seems like some bit is being set when steam is running that when it reloads it blows up on.

I was able to come up with pretty reliable repro though, made a docker image to contain the problem so I can repeat it. Even if you don't actually try this docker setup, point being it repros on a clean environment.

Dockerfile
```
FROM nvidia/opengl:1.0-glvnd-runtime-ubuntu18.04
RUN apt update
RUN apt install -y steam
```

Launch command
```
$ xhost +local:root
$ nvidia-docker run \
    --interactive --tty \
    --env="DISPLAY" \
    --env="NVIDIA_VISIBLE_DEVICES=all" \
    --env="NVIDIA_DRIVER_CAPABILITIES=all" \
    --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" \
    steam_image \
    /usr/games/steam && /usr/games/steam # second run will fail
```

**Comment by draeath** [comment]
@jamesdolan I would make a note of the image hash you are referencing in FROM (as a comment) so it's possible to notice a replaced upstream image with the same tag. This is important if reproducing something is the intent.

I've had this happen to me *cough* jetty *cough* - version numbers stayed the same but image was rebuilt and subtly different due to a different further-upstream image.

**Comment by lilianmoraru** [comment]
@cheloide 
> I recommend installing the [kernel headers](https://snapshot.debian.org/archive/debian/20190315T212803Z/pool/main/l/linux/linux-headers-4.19.0-4-amd64_4.19.28-2_amd64.deb) first if you have nvidia drivers.

To add to that(info for other users):
I recommend running `dkms status` first to see that the nvidia dkms module was compiled and installed correctly first.
You should see something like this:
```
nvidia, 430.40, 5.2.5, x86_64: installed
```
Instead of `5.2.5`, it should be your kernel version.

**Comment by Lyberta** [comment]
Getting the same error on Debian Testing/Bullseye with Linux 5.2.9.

Although it happens when I try to launch games. The game process just sits there for several minutes and then crashes.

**Comment by ignacioPL** [comment]
I have the same problem in Linux Mint with kernel 5.0.0-29 I also tried 4.18.0-25 and 4.15.0-64 all the same issue.

**Comment by kisak-valve** [comment]
As [previously noted](https://github.com/ValveSoftware/steam-for-linux/issues/6326#issuecomment-506124600), this issue was resolved with tcp: refine memory limit test in tcp_fragment() and was backported to the LTS kernels.

This has gotten into most distro-provided kernels and there's nothing left to investigate here.

I'm going to lock this issue to avoid spam going to the subscribed participants.

@ignacioPL, there is an issue with Steam's servers right now, not your system.
